src/views/Pools/components/PoolCard/Modals/StakeModal.tsx

```diff
@@ -43,7 +43,13 @@ const StakeModal: React.FC<StakeModalProps> = ({
   const [stakeAmount, setStakeAmount] = useState('')
   const [hasReachedStakeLimit, setHasReachedStakedLimit] = useState(false)
   const [percent, setPercent] = useState(0)
-  const calculatedStakingLimit = stakingTokenBalance.gt(stakingLimit) ? stakingLimit : stakingTokenBalance
+  const getCalculatedStakingLimit = () => {
+    if (isRemovingStake) {
+      return userData.stakedBalance
+    }
+    return stakingTokenBalance.gt(stakingLimit) && stakingLimit.gt(0) ? stakingLimit : stakingTokenBalance
```

Comment:
Added `&& stakingLimit.gt(0)` - to catch the majority of pools which will have a stakingLimit of 0

Reply:
`stakingLimit.gt(0)` should be the first clause so it bails out earlier. (micro optimization I know)

Reply:
Tasty. Will do